components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7156,6 +7156,12 @@ private void handleException(String msg, Throwable t) throws APIManagementExcept
         throw new APIManagementException(msg, t);
     }
 
+    private void handleException(String msg) throws APIManagementException {
+
+        log.error(msg);
+        throw new APIManagementException(msg);
```

Comment:
Don't log and throw please

Reply:
+1 to fix.
This is the same pattern used in the existing method override as well. https://github.com/wso2/carbon-apimgt/pull/12107/files#diff-ac8fc1b7b2d10089084f9f842ad03ac096b64f8c555798aa18a926e0ec663a4cR7153-R7157

Reply:
Removed this error handler method since the usages are updated to directly throw the APIManagementException.
Fix: Fixed with: https://github.com/wso2/carbon-apimgt/pull/12107/commits/054a492392c305d70924d72ad346694a7c75508e